File: scripts/00_utils/paths.py
"""
Shared path resolution -- everything computed relative to this file's own
location instead of hardcoded absolute paths, so the codebase runs the same
whether it's on this dev machine or a deployed container. Same pattern
already used correctly for MODEL_PATH in compare_swing.py, extended here
into one shared module other files import from instead of each redefining
their own relative walk-up.
"""
import os

SCRIPTS_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
REPO_ROOT   = os.path.dirname(SCRIPTS_DIR)
DATA_DIR    = os.path.join(REPO_ROOT, 'data')
BACKEND_DIR = os.path.join(REPO_ROOT, 'backend')

File: backend/src/config/paths.js
const path = require('path');

// Computed relative to this file's own location, same idea as
// scripts/00_utils/paths.py on the Python side -- one shared source of
// truth instead of every route file redefining its own relative walk-up.
const SCRIPTS_DIR = path.join(__dirname, '..', '..', '..', 'scripts');
const DATA_DIR = path.join(__dirname, '..', '..', '..', 'data');

// PYTHON_BIN env var is an escape hatch; otherwise pick the venv binary by
// platform so this works unchanged on Windows dev and a Linux container --
// venv layouts differ (Scripts/python.exe vs bin/python3).
const PYTHON = process.env.PYTHON_BIN || path.join(
  SCRIPTS_DIR, 'venv',
  process.platform === 'win32' ? 'Scripts/python.exe' : 'bin/python3'
);

module.exports = { SCRIPTS_DIR, DATA_DIR, PYTHON };
